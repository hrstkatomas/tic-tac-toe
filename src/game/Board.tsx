import React from 'react'
import Square from './Square'

const Board: React.FunctionComponent = () => {
  return (
    <div>
      <div>
        <Square />
        <Square />
        <Square />
      </div>
      <div>
        <Square />
        <Square />
        <Square />
      </div>
      <div>
        <Square />
        <Square />
        <Square />
      </div>
    </div>
  )
}

export default Board
