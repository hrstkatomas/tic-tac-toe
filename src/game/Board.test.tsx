import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Board from './Board'

describe("Board component", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Board />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders according to snapshot', () => {
    const tree = renderer
      .create(<Board />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
