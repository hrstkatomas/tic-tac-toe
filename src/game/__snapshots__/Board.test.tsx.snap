// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component renders according to snapshot 1`] = `
<div>
  <div>
    <button>
      x
    </button>
    <button>
      x
    </button>
    <button>
      x
    </button>
  </div>
  <div>
    <button>
      x
    </button>
    <button>
      x
    </button>
    <button>
      x
    </button>
  </div>
  <div>
    <button>
      x
    </button>
    <button>
      x
    </button>
    <button>
      x
    </button>
  </div>
</div>
`;
